glom/core.py

```diff
@@ -2531,3 +2537,38 @@ def FILL(target, spec, scope):
     if callable(spec):
         return spec(target)
     return spec
+
+class _ArgValuator(object):
+    def __init__(self):
+        self.cache = {}
+
+    def mode(self, target, spec, scope):
+        """
+        similar to FILL, but without function calling;
+        useful for default, scope assignment, call/invoke, etc
+        """
+        recur = lambda val: scope[glom](target, val, scope)
+        result = spec
+        if type(spec) in (list, dict):  # can contain themselves
+            if id(spec) in self.cache:
+                return self.cache[id(spec)]
+            result = self.cache[id(spec)] = type(spec)()
+            if type(spec) is dict:
+                result.update({recur(key): recur(val) for key, val in spec.items()})
+            else:
+                result.extend([recur(val) for val in spec])
+        if type(spec) in (tuple, set, frozenset):  # cannot contain themselves
```

Comment:
Just a thought, but how do we ensure that arg mode stays supporting the same set of containers as Fill? (dict, list, tuple, set, frozenset) Also, do we expect that will grow? Is it worth documenting them separately if they might diverge in future?

Reply:
my initial thought was "things that python has syntax support for" -- but then I went and included frozenset

Reply:
I don't expect this to grow -- its easy enough to embed your own callables into glom

in some sense, these data structures form the "syntax" of glom (nested tuples, lists, dicts)